elm/ords/process.py

```diff
@@ -0,0 +1,686 @@
+        the document's ``metadata`` attribute.
+    """
+    start_time = time.time()
+    doc = await download_county_ordinance(
+        county,
+        text_splitter,
+        num_urls=num_urls,
+        file_loader_kwargs=file_loader_kwargs,
+        browser_semaphore=browser_semaphore,
+        **kwargs,
+    )
+    if doc is None:
+        await _record_time_and_usage(start_time, **kwargs)
+        return None
+
+    doc.metadata["location"] = county
+    doc.metadata["location_name"] = county.full_name
+    await _record_usage(**kwargs)
+
+    doc = await extract_ordinance_text_with_ngram_validation(
+        doc, text_splitter, **kwargs
+    )
+    await _record_usage(**kwargs)
+
+    doc = await _write_cleaned_text(doc)
+    doc = await extract_ordinance_values(doc, **kwargs)
+
+    if _doc_contains_ord_values(doc):
+        doc = await _move_file_to_out_dir(doc)
+        doc = await _write_ord_db(doc)
+
+    await _record_time_and_usage(start_time, **kwargs)
+    return doc
+
+
+async def _record_usage(**kwargs):
+    """Dump usage to file if tracker found in kwargs."""
+    usage_tracker = kwargs.get("usage_tracker")
+    if usage_tracker:
+        await UsageUpdater.call(usage_tracker)
+
+
+async def _record_time_and_usage(start_time, **kwargs):
+    """Add elapsed time before updating usage to file."""
+    seconds_elapsed = time.time() - start_time
+    usage_tracker = kwargs.get("usage_tracker")
+    if usage_tracker:
+        usage_tracker["total_time_seconds"] = seconds_elapsed
+        usage_tracker["total_time"] = str(timedelta(seconds=seconds_elapsed))
+        await UsageUpdater.call(usage_tracker)
+
+
+async def _move_file_to_out_dir(doc):
+    """Move PDF or HTML text file to output directory."""
+    out_fp = await FileMover.call(doc)
+    doc.metadata["out_fp"] = out_fp
+    return doc
+
+
+async def _write_cleaned_text(doc):
+    """Write cleaned text to `clean_dir`."""
+    out_fp = await CleanedFileWriter.call(doc)
+    doc.metadata["cleaned_fp"] = out_fp
+    return doc
+
+
+async def _write_ord_db(doc):
+    """Write cleaned text to `county_dbs_dir`."""
+    out_fp = await OrdDBFileWriter.call(doc)
+    doc.metadata["ord_db_fp"] = out_fp
+    return doc
+
+
+def _setup_pytesseract(exe_fp):
+    """Set the pytesseract command."""
+    import pytesseract
+
+    logger.debug("Setting `tesseract_cmd` to %s", exe_fp)
+    pytesseract.pytesseract.tesseract_cmd = exe_fp
+
+
+def _record_total_time(fp, seconds_elapsed):
+    """Dump usage to an existing file."""
+    if not Path(fp).exists():
+        usage_info = {}
+    else:
+        with open(fp, "r") as fh:
+            usage_info = json.load(fh)
+
+    total_time_str = str(timedelta(seconds=seconds_elapsed))
+    usage_info["total_time_seconds"] = seconds_elapsed
+    usage_info["total_time"] = total_time_str
+
+    with open(fp, "w") as fh:
+        json.dump(usage_info, fh, indent=4)
+
+    logger.info("Total processing time: %s", total_time_str)
```

Comment:
Is there any way we could add an additional log message somewhere around here stating "{} ordinances found for {county} {state}. Outputs are here: {}"

Reply:
Yup! Added this and also a message if no ordinance values are found at all.